src/helpers.ts

```diff
@@ -48,6 +53,21 @@ export function generateContractSpecificId(
   return contractAddress.toHexString() + "-" + entityId.toString();
 }
 
+// returns new whitelisting id
+export function addWhitelisting(contractId: string, accountId: string): string {
+  let account = new Account(accountId);
+  account.save();
+
+  let whitelisting = new Whitelisting(
+    generateWhitelistingId(contractId, account.id)
+  );
+  whitelisting.account = account.id;
+  whitelisting.contract = contractId;
+
+  whitelisting.save();
+  return whitelisting.id;
```

Comment:
Any reason to not just return the full whitelisting entity here?

Reply:
good call! no reason - updated to return entity instead of entity id 👍